ext/hash/hash.c

```diff
@@ -361,18 +361,18 @@ PHP_FUNCTION(hash_init)
 
 	ops = php_hash_fetch_ops(algo);
 	if (!ops) {
-		zend_throw_error(NULL, "Unknown hashing algorithm: %s", ZSTR_VAL(algo));
+		zend_argument_value_error(1, "must be a valid hashing algorithm");
 		RETURN_THROWS();
 	}
 
 	if (options & PHP_HASH_HMAC) {
 		if (!ops->is_crypto) {
-			zend_throw_error(NULL, "HMAC requested with a non-cryptographic hashing algorithm: %s", ZSTR_VAL(algo));
+			zend_argument_value_error(2, "must not request HMAC with a non-cryptographic hashing algorithm");
```

Comment:
I'd rather shift this around to `argument #1 must be a cryptographic hashing algorithm if HMAC is requested`, like the other messages use.